django/contrib/admin/views/autocomplete.py

```diff
@@ -24,12 +25,19 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
+        self.to_field = request.GET.get('to_field', None)
+        if self.to_field:
+            if not self.model_admin.to_field_allowed(request, self.to_field):
+                raise DisallowedModelAdminToField("The field %s cannot be referenced." % self.to_field)
+        else:
+            self.to_field = 'pk'
+
         self.paginator_class = self.model_admin.paginator
-        self.object_list = self.get_queryset()
+        self.object_list = self.get_queryset().order_by(self.to_field)
```

Comment:
I'm ordering here because otherwise I get an `UnorderedObjectListWarning` in the `test_success` unit test
```
======================================================================
ERROR: test_success (admin_views.test_autocomplete_view.AutocompleteJsonViewTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/constantino/Desktop/django/tests/admin_views/test_autocomplete_view.py", line 73, in test_success
    response = AutocompleteJsonView.as_view(**self.as_view_args_to_field)(request)
  File "/Users/constantino/Desktop/django/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/constantino/Desktop/django/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/Users/constantino/Desktop/django/django/contrib/admin/views/autocomplete.py", line 38, in get
    context = self.get_context_data()
  File "/Users/constantino/Desktop/django/django/views/generic/list.py", line 119, in get_context_data
    paginator, page, queryset, is_paginated = self.paginate_queryset(queryset, page_size)
  File "/Users/constantino/Desktop/django/django/views/generic/list.py", line 58, in paginate_queryset
    allow_empty_first_page=self.get_allow_empty())
  File "/Users/constantino/Desktop/django/django/contrib/admin/views/autocomplete.py", line 49, in get_paginator
    return self.model_admin.get_paginator(self.request, *args, **kwargs)
  File "/Users/constantino/Desktop/django/django/contrib/admin/options.py", line 796, in get_paginator
    return self.paginator(queryset, per_page, orphans, allow_empty_first_page)
  File "/Users/constantino/Desktop/django/django/core/paginator.py", line 33, in __init__
    self._check_object_list_is_ordered()
  File "/Users/constantino/Desktop/django/django/core/paginator.py", line 125, in _check_object_list_is_ordered
    stacklevel=3
django.core.paginator.UnorderedObjectListWarning: Pagination may yield inconsistent results with an unordered object_list: <class 'admin_views.models.ReferencedByParent'> QuerySet.
```

Reply:
I think that's a different issue. I'd add a `Meta.ordering` to the test models to work around it for now.